src/metabase/lib/metadata/calculation.cljc

```diff
@@ -545,9 +545,9 @@
 
 (defn implicitly-joinable-columns
   "Columns that are implicitly joinable from some other columns in `column-metadatas`. To be joinable, the column has to
-  have appropriate FK metadata, i.e. have an `:fk-target-field-id` pointing to another Field. (I think we only include
-  this information for Databases that support FKs and joins, so I don't think we need to do an additional DB feature
-  check here.)
+  have (1) appropriate FK metadata, i.e. have an `:fk-target-field-id` pointing to another Field, and (2) have a numeric
+  `:id`, i.e. be a real database column that can be used in a JOIN condition. (I think we only include this information
+  for Databases that support FKs and joins, so I don't think we need to do an additional DB feature check here.)
```

Comment:
do we implicitly join compound keys?

Reply:
I'm not sure what you mean?

Reply:
for some reason I thought we had something like this already. https://github.com/metabase/metabase/issues/7335
Obviously, irrelevant to this issue at this point.